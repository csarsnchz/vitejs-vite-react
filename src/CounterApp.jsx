import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ valor }) => {
  const [counter, setCounter] = useState(valor);

  const handleAdd = () => {
    setCounter(counter + 1);
    //setCounter((c) => c + 1);
  };

  const handleSubstract = () => {
    setCounter(counter - 1);
    //setCounter((c) => c + 1);
  };

  const handleReset = () => {
    setCounter(valor);
    //setCounter((c) => c + 1);
  };
  return (
    <>
      <h1>Counter App</h1>
      <h2>{counter}</h2>

      <button onClick={handleAdd}>+1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleSubstract}>-1</button>
    </>
  );
};

CounterApp.propTypes = {
  valor: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
  valor: 0,
};
