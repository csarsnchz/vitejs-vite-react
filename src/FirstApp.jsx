import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <h3>{subTitle}</h3>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  name: PropTypes.string,
};

FirstApp.defaultProps = {
  subtitle: 'Testing',
  name: 'Csar',
};
