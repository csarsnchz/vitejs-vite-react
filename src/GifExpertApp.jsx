import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
export const GifExpertApp = () => {
  const [categories, setCategories] = useState([
    'Dragonball',
    'One Piece',
    'One Punch',
    'Samurai X',
  ]);
  const onAddCategory = () => {
    setCategories(['Valorant', ...categories]);
    //setCategories((cat) => [...cat, 'Voltron']);
  };
  return (
    <>
      {/* titulo */}
      <h1>GifExpertApp</h1>

      {/* Input */}
      <AddCategory />
      {/* Listado de Gif */}
      <button onClick={onAddCategory}>Agregar</button>
      <ol>
        {categories.map((category) => {
          return <li key={category}>{category}</li>;
        })}
      </ol>
      {/* Gif Item */}
    </>
  );
};
