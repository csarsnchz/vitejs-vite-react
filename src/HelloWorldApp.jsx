//export function App() {
//  return <h1>Hola Mundo!!!</h1>;
//}

const neWMessage = (valor) => {
  if (valor) {
    return (
      <>
        <h2>Esto es una funcion con param: </h2>
        <h3>{valor}</h3>
      </>
    );
  } else {
    return (
      <>
        <h2>Esto es una funcion sin param </h2>
      </>
    );
  }
};

export const HelloWorldApp = () => {
  return (
    <>
      <h1>{neWMessage()}</h1>
      <p>Subtitulo</p>
    </>
  );
};
