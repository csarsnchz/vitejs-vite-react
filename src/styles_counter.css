html,
body {
  background-color: darkslategray;
  color: aliceblue;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

button {
  border: 1;
  padding: 15px;
  font-size: 1.3rem;
  margin: 5px;
}
