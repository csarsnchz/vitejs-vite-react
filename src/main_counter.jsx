import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles_counter.css';
import { CounterApp } from './CounterApp';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp valor={100} />
  </React.StrictMode>
);
