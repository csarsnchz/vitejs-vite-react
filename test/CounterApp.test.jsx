import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';
describe('Pruebas en componente <CounterApp />', () => {
   const valor = 100
   test('Debe hacer match con la configuración de snapshot', () => {
     const { container } = render(<CounterApp valor={valor}/>);
     expect( container ).toMatchSnapshot();
   });
   test('Debe mostrar el valor en h2', () => {
     render(<CounterApp valor={valor}/>);
     expect( screen.getByText(valor)).toBeTruthy();
     expect( screen.getByRole('heading',{ level: 2 }).innerHTML).toContain('100');
  });
  test('Debe incrementar con el botón +1', () => {
    render(<CounterApp valor={valor}/>);
    fireEvent.click( screen.getByText('+1'));
    expect(screen.getByText('101')).toBeTruthy();
  });
  test('Debe decrementar con el botón -1', () => {
    render(<CounterApp valor={valor}/>);
    fireEvent.click( screen.getByText('-1'));
    expect(screen.getByText('99')).toBeTruthy();
  });
  test('Debe reiniciar el valor inicial', () => {
    render(<CounterApp valor={valor}/>);
    fireEvent.click( screen.getByText('+1'));
    fireEvent.click( screen.getByText('+1'));
    fireEvent.click( screen.getByText('+1'));
    fireEvent.click( screen.getByText('Reset'));
    expect(screen.getByText(valor)).toBeTruthy();
  });
});
