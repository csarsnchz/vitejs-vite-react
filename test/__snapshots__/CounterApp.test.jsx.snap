// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <CounterApp /> Debe hacer match con la configuraci√≥n de snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    100
  </h2>
  <button>
    +1
  </button>
  <button>
    Reset
  </button>
  <button>
    -1
  </button>
</div>
`;
